### PR TITLE
Converts the Cargo Bounties UI to TGUI and adds a modular app for it.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
How did bounty boards take me like 2 months but I finished this in 4 hours on a weekend, I'd be proud if I wasn't so baffled

## About The Pull Request

image
_Standard, standalone computer UI on the left, modular app UI on the right._
First and foremost, this reworks the cargo bounty UI to work within TGUI. Functionality is fully the same, but it means that there's no longer a need for a refresh button at the top right alongside the bounty sheet printer.
Second, while I was staring at this, I also ported it over to a modular computer app because it turned out to be a lot simpler than I imagined, considering it runs off of global lists.
Lastly, I went and pre-installed both this app as well as the other cargo-inspired app into the cargo tablets. For those installing the app instead, it requires cargo access to be downloaded, but it enables cargo crew to claim and read the list of bounties anywhere on the station.

## Why It's Good For The Game

I tried doing a project with the cargo bounty system a while back, but because of the un-responsiveness from the HTML UI, it was enabling all sorts of weird bugs and issues, and being a TGUI will help solve some problems down the road.
Also, this is a major part of what cargo can and should be doing during the shift, so having bounties available on a tablet increases the value of the modular app system as well. As an added bonus, now the cargo tablet has a reason to be used outside of "sell it for parts".

## Changelog
:cl:
add: Adds the Cargo Bounty UI as a modular app, with cargo access.
tweak: The Cargo Bounty UI has been revamped.
tweak: Cargo tablets now come preinstalled with the price tagger and cargo bounty apps.
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
I am so fucking rigid right now

How hard would it be to make a modular app for ordering shit from cargo?

**Comment by zxaber** [comment]
The requests console is already tgui, so probably not difficult at all.

**Review by Twaticus — APPROVED** [review]

**Review by stylemistake — DISMISSED** [review]

**Comment by ArcaneMusic** [comment]
Quick screenshot of the post updated visuals, High priority bounties are still highlighted.
image

File: tgui/packages/tgui/interfaces/CargoBountyConsole.js [hunk 1]
```diff
@@ -16,91 +16,110 @@ export const CargoBountyConsole = (props, context) => {
     <Window resizable>
       <Window.Content scrollable>
         <Section
+          title=<BountyHeader />
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/CargoBountyConsole.js:19]:
```suggestion
          title={<BountyHeader />}
```

File: tgui/packages/tgui/interfaces/NtosBountyConsole.js [hunk 2]
```diff
@@ -16,91 +16,109 @@ export const NtosBountyConsole = (props, context) => {
     <NtosWindow resizable>
       <NtosWindow.Content scrollable>
         <Section
+          title=<BountyHeader />
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/NtosBountyConsole.js:19]:
```suggestion
          title={<BountyHeader />}
```

File: code/modules/cargo/bounty_console.dm [hunk 3]
```diff
@@ -32,65 +32,35 @@
...
-	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
-	popup.open()
-
-/obj/machinery/computer/bounty/Topic(href, href_list)
+/obj/machinery/computer/bounty/ui_act(action,params)
 	if(..())
 		return
-
-	switch(href_list["choice"])
+	var/datum/bounty/cashmoney = locate(params["bounty"]) in GLOB.bounties_list
```
Comment (AnturK) [inline code/modules/cargo/bounty_console.dm:55]: Move this inside the case its for.

File: code/modules/cargo/bounty_console.dm [hunk 4]
```diff
@@ -32,65 +32,35 @@
...
-			var/datum/bounty/B = locate(href_list["d_rec"]) in GLOB.bounties_list
-			if(B)
-				B.claim()
-
-	if(href_list["refresh"])
-		playsound(src, "terminal_type", 25, FALSE)
-
-	updateUsrDialog()
+				return
+	. = TRUE
```
Comment (AnturK) [inline code/modules/cargo/bounty_console.dm:66]: Why are you updating if there was no changes. This should only return true for claim here from what i can see.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/CargoBountyConsole.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    bountydata = [],
+    stored_cash,
+  } = data;
+  return (
+    <Window resizable>
+      <Window.Content scrollable>
+        <Section
+          title={<BountyHeader />}
+          buttons={(
+            <Flex>
```
Comment (AnturK) [inline tgui/packages/tgui/interfaces/CargoBountyConsole.js:21]: Why Flex if there's only a single element

**Pull request merged**